doc/aaa/TACACS+ Server Monitor.md

```diff
@@ -0,0 +1,86 @@
+# TACACS+ server monitor design
+
+## Overview
+
+SONiC device usually configured with multiple TACACS+ server, when a server is unreachable, SONiC device will try to connect with next TACACS+ server.
+
+SONiC device will communicate with TACACS+ server in following scenarios:
+1. Remote user login to SONiC device.
+2. Remote user run commands on SONiC device.
+
+There is a timeout for each server, the default value is 5 seconds, this means if the first server not reachable, SONiC device will stuck there when user login or running commands.
+
+To improve this issue, SONiC will add a TACACS+ server monitor to change server priority, a server unreachable or slow response will be downgrade.
+
+### Functional Requirement
+- Monit TACACS+ server unreachable event from COUNTER_DB.
+- Monit TACACS+ server slow response event from COUNTER_DB.
+- Change server priority based unreachable event and slow response event.
+- Not change any other server attribute.
+- Not change any other TACACS+ config.
+
+### Counter DB schema
+#### TACPLUS_SERVER_LATENCY Table schema
+```
+; Key
+server_key           = IPAddress  ;  TACACS+ server’s address
+; Attributes
+latency              = 1*10DIGIT  ; server network latency in MS, -1 for connect to server timeout
+```
+
+### Config DB schema
+#### TACPLUS_SERVER_MONITOR Table schema
```

Comment:
Would it make sense to rename this to TACPLUS_MONITOR (there is only one row in this table 'config') ?

Reply:
Fixed.
Because every table must have at least 1 row, so it's common in CONFIG_DB have 1 row tables for config.